src/praisonai-agents/praisonaiagents/agent/handoff.py

```
@@ -924,6 +932,391 @@ async def _do_handoff():
+                duration_seconds=time.time() - start_time,
+                error=f"Timeout after {self.config.timeout_seconds}s",
+                handoff_depth=_get_handoff_depth(),
+            )
+            self._execute_callback(self.config.on_error, source_agent, kwargs, result)
+            raise HandoffTimeoutError(
+                f"Typed handoff to {self.agent.name} timed out after {self.config.timeout_seconds}s",
+                timeout_seconds=self.config.timeout_seconds,
+                source_agent=source_agent.name,
+                target_agent=self.agent.name,
+                agent_id=source_agent.name
+            )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)